src/v/redpanda/application.cc

```diff
@@ -430,8 +431,12 @@ int application::run(int ac, char** av) {
     // use endl for explicit flushing
     std::cout << community_msg << std::endl;
 
-    return app.run(ac, av, [this, &app] {
+    // NOLINTNEXTLINE: cppcoreguidelines-pro-bounds-pointer-arithmetic
+    std::string cmd_line = fmt::to_string(fmt::join(av, av + ac, " "));
```

Comment:
this form it's longer but should not trigger the warning 
```suggestion
    std::string cmd_line = fmt::to_string(fmt::join(std::span{av, size_t(ac)}, " "));
```

Reply:
Except for warning this just seems worse, but let's do it.